aggsender/query/l1info_tree_data_query.go

```diff
@@ -77,27 +76,41 @@ func (l *L1InfoTreeDataQuerier) GetL1InfoRootByLeafIndex(ctx context.Context,
 	return &root, nil
 }
 
-// GetFinalizedL1InfoTreeData returns the L1 Info tree data for the last finalized processed block
-// l1InfoTreeData is:
-// - merkle proof of given l1 info tree leaf
-// - the leaf data of the highest index leaf on that block and root
-// - the root of the l1 info tree on that block
-func (l *L1InfoTreeDataQuerier) GetFinalizedL1InfoTreeData(ctx context.Context,
-) (treetypes.Proof, *l1infotreesync.L1InfoTreeLeaf, *treetypes.Root, error) {
-	root, leaf, err := l.GetLatestFinalizedL1InfoRoot(ctx)
+// GetFinalizedL1InfoTreeData retrieves the L1 info tree leaf and its merkle proof for a finalized L1 info tree state.
+// It takes the finalized L1 info tree root hash and leaf count to fetch the last leaf in the tree
+// and generate a merkle proof from that leaf to the specified root hash.
+//
+// Parameters:
+//   - ctx: Context for cancellation and timeouts
+//   - finalizedL1InfoTreeRootHash: The root hash of the finalized L1 info tree
+//   - finalizedL1InfoTreeLeafCount: The total number of leaves in the finalized L1 info tree
+//
+// Returns:
+//   - treetypes.Proof: The merkle proof from the leaf to the root
+//   - *l1infotreesync.L1InfoTreeLeaf: The last leaf in the finalized tree
+//   - error: Any error that occurred during the operation
+func (l *L1InfoTreeDataQuerier) GetFinalizedL1InfoTreeData(
+	ctx context.Context,
+	finalizedL1InfoTreeRootHash common.Hash,
+	finalizedL1InfoTreeLeafCount uint32,
+) (treetypes.Proof, *l1infotreesync.L1InfoTreeLeaf, error) {
+	leafIndex := finalizedL1InfoTreeLeafCount - 1
+
+	leaf, err := l.GetInfoByIndex(ctx, leafIndex)
 	if err != nil {
-		return treetypes.Proof{}, nil, nil,
-			fmt.Errorf("error getting latest finalized L1 Info tree root: %w", err)
+		return treetypes.Proof{}, nil,
+			fmt.Errorf("error getting L1 Info tree leaf by index %d: %w", leafIndex, err)
 	}
 
-	proof, err := l.l1InfoTreeSyncer.GetL1InfoTreeMerkleProofFromIndexToRoot(ctx, root.Index, root.Hash)
+	proof, err := l.l1InfoTreeSyncer.GetL1InfoTreeMerkleProofFromIndexToRoot(ctx,
+		finalizedL1InfoTreeLeafCount-1, finalizedL1InfoTreeRootHash)
```

Comment:
Why don't you reuse the leafIndex here?

```suggestion
		leafIndex, finalizedL1InfoTreeRootHash)
```

Reply:
Fix: https://github.com/agglayer/aggkit/pull/1264/commits/b8a346fdbddab3d94e8bda2a1790612ed4c398e4